packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-radio-group/gux-form-field-radio-group.tsx

```diff
@@ -0,0 +1,141 @@
+import { Component, Element, h, JSX, State, Prop, Watch } from '@stencil/core';
+
+import { OnMutation } from '@utils/decorator/on-mutation';
+import { hasSlot } from '@utils/dom/has-slot';
+
+import {
+  GuxFormFieldError,
+  GuxFormFieldHelp,
+  GuxFormFieldLegendLabel,
+  GuxFormFieldFieldsetContainer
+} from '../../functional-components/functional-components';
+import { getSlotTextContent } from '@utils/dom/get-slot-text-content';
+
+import { trackComponent } from '@utils/tracking/usage';
+
+/**
+ * @slot input - Required slot for input tag
+ * @slot label - Required slot for label tag
+ * @slot groupError - Optional slot for groupError message
+ * @slot groupHelp - Optional slot for groupHelp message
+ */
+@Component({
+  styleUrl: 'gux-form-field-radio-group.scss',
+  tag: 'gux-form-field-radio-group',
+  shadow: true
+})
+export class GuxFormFieldRadio {
+  private disabledObserver: MutationObserver;
+  private label: HTMLLabelElement;
+
+  @Element()
+  private root: HTMLElement;
+
+  /**
+   *  radio group has error text.
+   */
+  @State()
+  private hasGroupError: boolean = false;
+
+  /**
+   *  radio group has help text.
+   */
+  @State()
+  private hasGroupHelp: boolean = false;
+
+  /**
+   * Disables the radio buttons in the group.
+   */
+  @Prop()
+  disabled: boolean = false;
+
+  @Watch('hasGroupError')
+  watchGroupError(hasGroupError: boolean): void {
```

Comment:
This function and `setDisabledRadio` sort of have the same functionality just different properties I wonder could they be combined into one. I think the `form-field-checkbox-group` may have similar functionality so maybe a utility could be created potentially like , 

`private updateGroupProperties(selector: string, propName: string, propValue: boolean): void {
    const elements = this.root.querySelectorAll(selector);
    elements.forEach(element => {
        element[propName] = propValue;
    });
}`

Reply:
I think this is a fair suggestion, but since the group error is determined by the presence of an error `slot` and the group disabled is set by the disabled prop I think it might be clearer from a readability standpoint to just have the functions be separate.